pkg/asset/handler/get.go

```diff
@@ -115,19 +115,19 @@ func (h *GetHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			return nil
 		}
 
-		// Remove existing CORS header in response
-		for name := range w.Header() {
-			if strings.HasPrefix(name, "Access-Control-") {
-				w.Header().Del(name)
-			}
-		}
-
 		resp.Header = h.CloudStorageProvider.ProprietaryToStandard(resp.Header)
 		// Do not support range request if image processing query is present.
 		if hasPipeline {
 			resp.Header.Del("Accept-Ranges")
 		}
 
+		// Delete existing CORS header if exists in response
+		for name := range resp.Header {
```

Comment:
If `Access-Control-` is ever found in upstream response, we must remove all of our CORS headers. The rationale is that if the upstream has any CORS headers, then we treat them as complete and do not merge with ours.